torch/distributed/algorithms/ddp_comm_hooks/powerSGD_hook.py

```diff
@@ -92,25 +95,25 @@ def powerSGD_hook(
     input_tensor[total_length:padded_total_length].fill_(0)
     matrix = input_tensor.view(square_side_length, square_side_length)
 
-    def create_low_rank_tensor(fill_random_values):
+    def create_low_rank_tensor(fill_random_values, rng):
         "Returns a low-rank 2D tensor of square_side_length * matrix_approximation_rank."
         if fill_random_values:
-            with torch.random.fork_rng(devices=[device]):
+            with torch.random.fork_rng(devices=[]):
```

Comment:
Could you add a comment here on why we need to fork_rng (to ensure that the manual_seed is not applied globally)?